app/services/proxy_status/tftp.rb

```diff
@@ -0,0 +1,21 @@
+module ProxyStatus
+  class TFTP < Base
+    def initialize(proxy, opts = {})
+      super(proxy, opts, 'TFTP')
+    end
+
+    def tftp_server
+      raise ::Foreman::Exception.new(N_('No TFTP feature for %s'), proxy.to_label) unless proxy.has_feature?('TFTP')
```

Comment:
This would mean, I'd have to rescue on the controller action (where I call `.new`), which I'm less happy to do. I think that initializing should succeed and fail when calling the `tftp_server` method.
What do you think?